File: Text/fizz_buzz/fizz_buzz.rs
fn main() {
    for num in 1..101 {
      if (num % 3 == 0) && (num % 5 == 0) {
        println!("FizzBuzz");
      } else if num % 3 == 0 {
        println!("Fizz");
      } else if num % 5 == 0 {
        println!("Buzz");
      } else {
        println!("{}", num);
      }
    }
}

File: Text/fizz_buzz/fizz_buzz.kt
fun main(){
    //Direct usage
    0.fizzBuzz()

    //An input usage
    println("Enter a number to start FizzBuzz")
    (readLine()?.takeIf {it.isNotEmpty() && it.all(Char::isDigit)}?.toInt() ?: 0).fizzBuzz()
}

fun Int.fizzBuzz(){
    for (i in 0..this){
        println(when {
            i % 15 == 0 -> "FizzBuzz"
            i % 3  == 0 -> "Fizz"
            i % 5  == 0 -> "Buzz"
            else -> i
        })
    }
}
